SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+		$autoUnlockIntervalSec = $products.productAttributes.passwdUnlockIntervalInSec
+		$failedAttemptIntervalSec = $products.productAttributes.passwdAttemptsIntervalInSec
+		$numAttempts = $products.productAttributes.passwdMaxFailAttempts
+		Write-LogMessage -type INFO -Message "Setting Password Policy on vCenter SSO services.  Starting..."
+		Start-Sleep -s 5
+		#Write-Host "Connect to sso admin server" -ForegroundColor Yellow
+		if ($ppmVariables.ssoServerFqdn -and $ppmVariables.ssoServerUser -and $ppmVariables.ssoServerPass) {
+			Connect-SsoAdminServer -Server $ppmVariables.ssoServerFqdn -User $ppmVariables.ssoServerUser -Password $ppmVariables.ssoServerPass | Out-Null
+		} else {
+			Connect-SsoAdminServer -server $vcenter.fqdn -user $vcenter.ssoAdmin -pass $vcenter.ssoAdminPass
+		}
+		#Write-Host "Set sso password policy" -ForegroundColor Yellow
+		Get-SsoPasswordPolicy | Set-SsoPasswordPolicy -ProhibitedPreviousPasswordsCount $passHistory -MinLength $minCharacterLength -MaxLength $maxCharacterLength -MinNumericCount $minNumericChar -MinSpecialCharCount $minSpecialChar -MaxIdenticalAdjacentCharacters $maxConsecutiveIdenticalChar -MinAlphabeticCount $minAlphabeticCount -MinUppercaseCount $minUppercaseChar -MinLowercaseCount $minLowercaseChar -PasswordLifetimeDays $passwordTtlInDays | Out-Null
+		#Write-Host "Set sso lockout" -ForegroundColor Yellow
+		Get-SsoLockoutPolicy | Set-SsoLockoutPolicy -AutoUnlockIntervalSec $autoUnlockIntervalSec -FailedAttemptIntervalSec $failedAttemptIntervalSec -MaxFailedAttempts $numAttempts | Out-Null
+		#Write-Host "Disconnect from sso admin server" -ForegroundColor Yellow
+		if($ppmVariables.ssoServerFqdn){
+			Disconnect-SsoAdminServer -Server $ppmVariables.ssoServerFqdn | Out-Null
+		} else {
+			Disconnect-SsoAdminServer -Server $vcenter.fqdn | Out-Null
+		}
+		Write-LogMessage -type INFO -Message "Setting Password Policy on vCenter SSO services.  Complete..."
+	}
+	
+	#ESXi
+	if ($ppmEnvironmentalDetails | where { ($_.productType -match "ESXi") -and ($_.productAttributes.driftAlarm -match "Red") } ) {
+		$products = $ppmStandardConfigValues | Select-Object -Skip 0 | Where-Object -FilterScript { $_.productType -eq "ESXi" }
+		$passwdExpInDays = $products.productAttributes.passwdExpInDays
+		$numAttempts = $products.productAttributes.passwdMaxFailAttempts
+		$for1CharClass = $products.productAttributes.passwdMinimumLengthFor1CharClass
+		$for2CharClass = $products.productAttributes.passwdMinimumLengthFor2CharClass
+		$for3CharClass = $products.productAttributes.passwdMinimumLengthFor3CharClass
+		$for4CharClass = $products.productAttributes.passwdMinimumLengthFor4CharClass
+		$forPhrases = $products.productAttributes.passwdMinimumCharLengthForPhrase
+		$policy = "retry=" + $numAttempts + " min=" + $for1CharClass + "," + $for2CharClass + "," + $forPhrases + "," + $for3CharClass + "," + $for4CharClass
+		Write-LogMessage -type INFO -Message "Setting ESXi hosts password policy.  Starting...."
```

Comment:
Change to: 'Setting Password Policy on ESXi hosts'

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e